src/gen.cpp

```diff
@@ -520,15 +522,26 @@ int generate_cmake(const char *path, bool root) {
         }
     }
 
-    if (!cmake.vcpkg.version.empty()) {
-        assert("pmm is required in fetch-content for vcpkg to work" && cmake.contents.count("pmm") != 0);
-        comment("Bootstrap vcpkg");
-        cmd("include")("${pmm_SOURCE_DIR}/pmm.cmake");
-        tsl::ordered_map<std::string, std::vector<std::string>> vcpkg_args;
-        vcpkg_args["REVISION"] = {cmake.vcpkg.version};
-        vcpkg_args["REQUIRES"] = cmake.vcpkg.packages;
-        auto vcpkg = std::make_pair("VCPKG", vcpkg_args);
-        cmd("pmm")(vcpkg).endl();
+    if (!cmake.vcpkg.packages.empty()) {
+        cmd("include")("FetchContent");
+        cmd("message")("STATUS", "Fetching vcpkg...");
+        cmd("FetchContent_Declare")("vcpkg", "URL", "https://github.com/microsoft/vcpkg/archive/refs/tags/2021.05.12.tar.gz");
+        cmd("FetchContent_MakeAvailable")("vcpkg");
+        cmd("include")("${vcpkg_SOURCE_DIR}/scripts/buildsystems/vcpkg.cmake");
+        using namespace nlohmann;
+        json j;
+        j["$schema"] = "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json";
+        j["name"] = cmake.project_name;
```

Comment:
According to https://vcpkg.readthedocs.io/en/latest/specifications/manifests/#definitions there are a bunch of requirements on the identifier:

- `<identifier>`: A `string` which:
  - Is entirely ASCII
  - Contains only lowercase alphabetic characters, digits, and hyphen-minus
  - Does not have multiple consecutive hyphens
  - Does not begin nor end with a hyphen
  - Is not a Windows filesystem reserved name
  - Is not a vcpkg reserved name: "default" or "core".
  - In other words, it must follow the regex `[a-z0-9]+(-[a-z0-9]+)*`, and must not be any of:
`{ prn, aux, nul, con, lpt[1-9], com[1-9], core, default }`

I would propose a `vcpkg_escape_identifier` function that generates a correct name

Reply:
I see. I've added a check but it generates a temporary name to avoid throwing.